quix-frontend/client/src/lib/runner/services/autocomplete/autocomplete-service.ts

```diff
@@ -46,39 +55,60 @@ export async function setupCompleters(
       query,
       position
     );
+
     const contextCompletions: Promise<AceCompletion[]> = sqlAutocompleter.getCompletionItemsFromQueryContext(
       queryContext
     );
 
     const [keywords, completions] = await Promise.all([
       keywordsCompletions,
-      contextCompletions,
+      contextCompletions
     ]);
 
-    let all =
-      queryContext.contextType === ContextType.Undefined
-        ? keywords
-        : completions;
+    const filteredCompletions: object[] = completions.filter(obj => obj.value.includes(prefix));
+    const searchInObject: Boolean = filteredCompletions.length === 0;
+    if (searchInObject) {
+      nestedColumnCompletions = sqlAutocompleter.getCompletionItemsFromQueryContextColumn(
+        queryContext
+      );
+      [nestedColumnCompletions] = await Promise.all([nestedColumnCompletions]);
+    }
 
-    if (prefix) {
-      const lowerCasedPrefix = prefix.trim().toLowerCase();
+    if (searchInObject && nestedColumnCompletions) {
+      all = queryContext.contextType === ContextType.Undefined ? keywords : nestedColumnCompletions;
+    } else {
+      all = completions;
+    }
 
-      all = all.reduce((resultArr: AceCompletion[], completionItem) => {
-        const indexes = findAllIndexOf(completionItem.value, lowerCasedPrefix);
 
-        if (indexes.length > 0) {
-          completionItem.matchMask = createMatchMask(
-            indexes,
-            lowerCasedPrefix.length
-          );
-          completionItem.score = 10000 - indexes[0];
-          resultArr.push(completionItem);
+    if (prefix) {
+      const lowerCasedPrefix = prefix.trim().toLowerCase();
+      if (searchInObject) {
+        if (prefix.endsWith('.')) {
+          all = enrichCompletionItemAfterDotObject(all);
         }
-
-        return resultArr;
-      }, []);
+        else {
+          all = queryContext.contextType === ContextType.Undefined
+            ? enrichCompletionKeyWord(all, lowerCasedPrefix)
+            : enrichCompletionItemInObjectSearch(all, queryContext, lowerCasedPrefix);
+        }
+      }
+      else {
+        const filterCompletions = all.filter(obj => obj.value.includes(prefix));
+        all = enrichCompletionItemCollumSearch(filterCompletions, lowerCasedPrefix)
+      }
     }
 
+    all.forEach(obj => {
+      const maxLengthOfString = 60;
+      const maxSubStringLength = 57;
+      if (obj.caption?.length > maxLengthOfString) {
+        obj.caption = obj.caption.substring(0, maxSubStringLength) + "..."
+      }
+      if (!obj.caption && obj.value.length > maxLengthOfString) {
+        obj.caption = obj.value.substring(0, maxSubStringLength) + "..."
+      }
```

Comment:
```suggestion
    obj.caption = createCaption(obj.caption || obj.value)
```

Reply:
replcae it with the new function you created..

consider to remove the all.forEach and add this part to updateCompletionItem

Reply:
got it

Reply:
very clean